js/main.js

```diff
@@ -20,14 +30,82 @@ function handleSubmit(event) {
   event.preventDefault();
   const newFormEntry = {
     entryId: data.nextEntryId,
-    nextEntryId: data.nextEntryId + 1,
     title: $entryTitle.value,
     photoURL: $photoURL.value,
     notes: $entryNotes.value,
   };
   data.nextEntryId += 1;
   data.entries.unshift(newFormEntry);
+  if (data.entries.length > 0) {
+    toggleNoEntries();
+  }
+  window.location.reload();
   $img?.setAttribute('src', 'images/placeholder-image-square.jpg');
   $form?.reset();
   writeData();
+  renderEntry(data.entries[data.nextEntryId]);
+  viewSwap('entries');
 }
+document.addEventListener('DOMContentLoaded', () => {
+  const storedData = readData();
+  if (storedData !== undefined) {
+    for (let i = 0; i < storedData.length; i++) {
+      $ul.append(renderEntry(storedData[i]));
+    }
+  }
+  if (data.entries.length > 0) {
+    toggleNoEntries();
+  }
+  viewSwap(readViewData());
+});
+function renderEntry(entry) {
+  const $list = document.createElement('li');
+  const $divRow = document.createElement('div');
+  $divRow.className = 'row';
+  const $divColumnHalfPhoto = document.createElement('div');
+  $divColumnHalfPhoto.className = 'column-half entry-photo';
+  const $divColumnHalfText = document.createElement('div');
+  $divColumnHalfText.className = 'column-half entry-text';
+  const $entryImg = document.createElement('img');
+  $entryImg.src = entry.photoURL;
+  const $entryHeader = document.createElement('h3');
+  $entryHeader.textContent = entry.title;
+  const $p = document.createElement('p');
+  $p.textContent = entry.notes;
+  $ul.append($list);
+  $list.append($divRow);
+  $divRow.append($divColumnHalfPhoto);
+  $divColumnHalfPhoto.append($entryImg);
+  $divRow.append($divColumnHalfText);
+  $divColumnHalfText.append($entryHeader);
+  $divColumnHalfText.append($p);
+  return $list;
+}
+function toggleNoEntries() {
+  $liNoEntries.className = 'hidden';
+}
```

Comment:
Inside of the function is where you should be checking if `data.entries` has a length, and you should not only be applying hidden, you should also be applying the opposite if there are no entries